test/framework/controlplane_helpers.go

```diff
@@ -154,17 +156,21 @@ type WaitForControlPlaneToBeReadyInput struct {
 // WaitForControlPlaneToBeReady will wait for a control plane to be ready.
 func WaitForControlPlaneToBeReady(ctx context.Context, input WaitForControlPlaneToBeReadyInput, intervals ...interface{}) {
 	By("Waiting for the control plane to be ready")
-	Eventually(func() (bool, error) {
-		controlplane := &controlplanev1.KubeadmControlPlane{}
+	controlplane := &controlplanev1.KubeadmControlPlane{}
+	Eventually(func() (controlplanev1.KubeadmControlPlane, error) {
 		key := client.ObjectKey{
 			Namespace: input.ControlPlane.GetNamespace(),
 			Name:      input.ControlPlane.GetName(),
 		}
 		if err := input.Getter.Get(ctx, key, controlplane); err != nil {
-			return false, err
+			return *controlplane, fmt.Errorf("failed to get KCP: %v", err)
```

Comment:
```suggestion
			return *controlplane, errors.Wrapf(err, "failed to get KCP")
```
nit, Sorry last one :). We usually use errors.Wrapf to wrap errors.

Reply:
The Gomega docs suggest that this error is actually ignored: https://onsi.github.io/gomega/#making-asynchronous-assertions

```
If multple values are returned by the function, Eventually will pass the first value to the matcher and require that all others are zero-valued. This allows you to pass Eventually a function that returns a value and an error - a common pattern in Go.
```

So I think you might as well just return an unwrapped error here. The interesting error message is with the PrettyPrint on line 173.

If we wanted to see this error message I think we might have to log it?

Reply:
Please see: https://github.com/kubernetes-sigs/cluster-api/pull/5601#discussion_r745114767 ++
It looks to me like it prints the error

Reply:
Done. Thanks for the reviews. 

The output will log an error when `err != nil`.
```
  Error: Unexpected non-nil/non-zero extra argument at index 1:
  	<*errors.withStack>: failed to get KCP: this is a sample error message that should show up
```